src/mlpack/core/data/is_naninf.hpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file is_naninf.hpp
+ * @author Ryan Curtin
+ *
+ * This is an adapted version of Conrad Sanderson's implementation of
+ * arma::diskio::convert_naninf() from Armadillo.  It is here so as to avoid
+ * using Armadillo internal functionality.
+ */
+#ifndef MLPACK_CORE_DATA_HAS_NANINF_HPP
+#define MLPACK_CORE_DATA_HAS_NANINF_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * See if the token is a NaN or an Inf, and if so, set the value accordingly and
+ * return a boolean representing whether or not it is.
+ */
+template<typename T>
+inline bool IsNaNInf(T& val, const std::string& token)
+{
+  // See if the token represents a NaN or Inf.
+  if ((token.length() == 3) || (token.length() == 4))
+  {
+    const bool neg = (token[0] == '-');
+    const bool pos = (token[0] == '+');
+
+    const size_t offset = ((neg || pos) && (token.length() == 4)) ? 1 : 0;
+
+    const std::string token2 = token.substr(offset, 3);
+
+    if ((token2 == "inf") || (token2 == "Inf") || (token2 == "INF"))
```

Comment:
I think this is faster as using: ``boost::algorithm::to_lower(str);`` and compare once; not sure.

Reply:
I'll actually leave this one as-is so that it replicates Armadillo's functionality exactly.  When 8.400 comes out, I may update this function with whatever Armadillo is doing and then use some ifdefs to pick the same one as the Armadillo version.